### PR TITLE
[Cherry-Pick] Adding AzurePolicyMode to Installation (#3618)

### PR DESCRIPTION
**Pull request by vikastigera**
* Adding AzurePolicyMode to Installationx

Changes done to add Azure config to installation spec which has option to provide PolicyMode. This helps clients to have control over "control-plane" labels for namespaces.

* Fixing merge failure

* fixing test

* Updating documentation

## Description
cherry pick pf https://github.com/tigera/operator/pull/3618

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
